### DISCUSSION TITLE
Is there a way to have multiple instances of ketcher-react open at the same time?

### DISCUSSION THREAD

**Discussion by alsersan** [epam/ketcher#5157] [Q&A]
Hello! First of all, thanks for all the effort put into this library.

So, I managed to successfully integrate Ketcher with my NextJS application, using ketcher-react. However, since ketcher uses the window object, I find it impossible to have multiple instances open at the same time. There are conflicts between instances (all fighting for the same window.ketcher object), and when I set a molecule in each instance, obviously only one instance successfully shows a molecule (which might even be from another instance).

I intend to create a "visualizer", where I just set the molecule and display it to the user (no edition), and I need multiple visualizers open at the same time, each with a different molecule. I would like to avoid using iframes, hence my initial question: is there any workaround? Or I'm just limited to one open instance at a time?

I found this open discussion ([https://github.com/epam/ketcher/discussions/2313](https://github.com/epam/ketcher/discussions/2313)), but it was not helpful, since ketcher expects a window.ketcher object (not window.whatever), and not passing it results in the following errors: 
![image](https://github.com/user-attachments/assets/dc54582d-8617-46e7-b66f-59d3548c7eeb)

I just want to be able to set a different molecule on each ketcher instance on initialization.

Thanks a lot in advance!

**Comment by rrodionov91** [comment]
Hi @alsersan 
Thank you for highlighting the issue.
We reproduced it and created ticket https://github.com/epam/ketcher/issues/5591.
